lib/checkother.cpp

```diff
@@ -1418,7 +1426,9 @@ void CheckOther::checkConstVariable()
             continue;
         if (var->isVolatile())
             continue;
-        if (isAliased(var))
+        const Token* alias = isAliased(var);
```

Comment:
This only checks the first alias found. You would need to check all aliases. However, I think this can be removed since `isExpressionChangedAt`(which is used by `isVariableChanged`) checks if aliases are modified.

Reply:
But we need to check for aliases in the return though if it escapes.

Reply:
We have this test case which we should warn since it returns by value and not by reference:

```cpp
auto foo(std::vector<int>& vec, bool flag) {
    std::vector<int> dummy;
    std::vector<int>::iterator iter;
    if (flag)
        iter = vec.begin();
    else {
        dummy.push_back(42);
        iter = dummy.begin();
    }
    return *iter;
}
```

Changing it `auto&`, cppcheck will correctly be silent(since `vec` cannot be `const`).